src/settings/HRIDHandling/HRIDHandlingSettings.test.js

```diff
@@ -0,0 +1,82 @@
+import {
+  screen,
+  fireEvent,
+  waitFor,
+} from '@folio/jest-config-stripes/testing-library/react';
+
+import '../../../test/jest/__mock__';
+import { renderWithIntl } from '../../../test/jest/helpers';
+import renderWithRouter from '../../../test/jest/helpers/renderWithRouter';
+
+import HRIDHandlingSettings from './HRIDHandlingSettings';
+
+
+
+const initialResources = {
+  hridSettings: {
+    records: [
+      {
+        commonRetainLeadingZeroes: true,
+        instance: { startNumber: '00000000001', prefix: '' },
+        holding: { startNumber: '00000000001', prefix: '' },
+        item: { startNumber: '00000000001', prefix: '' },
+      },
+    ],
+  },
+};
+
+const mockMutator = {
+  hridSettings: {
+    PUT: jest.fn(() => Promise.resolve()),
+  },
+};
+
+const renderComponent = (resources = initialResources, mutator = mockMutator) => {
+  return renderWithIntl(renderWithRouter(
+    <HRIDHandlingSettings resources={resources} mutator={mutator} />
+  ));
+};
+
+describe('HRIDHandlingSettings', () => {
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('renders without crashing', () => {
+    renderComponent();
+    expect(screen.getByText(/ViewMetaData/)).toBeInTheDocument();
+  });
+
+  it('shows confirmation modal on form submission', async () => {
+    renderComponent();
+
+    fireEvent.click(screen.getByText('confirm'));
+
+    await waitFor(() => expect(mockMutator.hridSettings.PUT).toHaveBeenCalled());
+  });
+
+  it('validates required fields', async () => {
+    renderComponent();
+
+    fireEvent.click(screen.getByRole('button', { name:'cancel' }));
+
+    await waitFor(() => expect(mockMutator.hridSettings.PUT).not.toHaveBeenCalled());
+  });
```

Comment:
I also do not see here checking the validation of some required fields. It is better to do something like focus and blur some fields and check that the validation error appears

Reply:
Updated description